monix-logback-core/src/main/scala/com/softwaremill/correlator/CorrelationId.scala

```diff
@@ -33,19 +27,17 @@ class CorrelationId(
 
   def applySync(): Option[String] = Option(MDC.get(MdcKey))
 
-  def setCorrelationIdMiddleware(service: HttpRoutes[Task]): HttpRoutes[Task] = Kleisli { req: Request[Task] =>
-    val cid = req.headers.get(CaseInsensitiveString(headerName)) match {
-      case None            => newCorrelationId()
-      case Some(cidHeader) => cidHeader.value
-    }
+  def setCorrelationIdMiddleware[T, R](
```

Comment:
maybe `withCorrelationId`? `Middleware` is a http-related term, and here we have nothing that's http-related.